[TEST] Make the BilinearInterpolation test deterministic

pjones opened the pull request:
## Description

The `BilinearInterpolation` test did not properly initialize all of the matrix elements before comparing for equality.  Since `NaN` does not compare with itself, the test would randomly fail when one or more elements were initialized to `NaN`.

I also removed unused headers as reported by clangd.

## Checklist
- [X] Make sure that you are listed in the AUTHORS file
- [X] New and existing unit tests pass locally with my changes

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Tests**
	- Disabled a previously failing test for the data copying functionality, replacing it with a directive to skip the test.
- **Chores**
	- Updated documentation settings to generate shorter file names for improved compatibility with certain file systems.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cbielow commented:
which part is actually failing?
Usually there is a subtle, yet potentially important bug hiding somewhere?!

cbielow commented:
this one?
https://cdash.seqan.de/test/212728848

cbielow commented:
well, that's "easy" to fix.
It seems if you do not initialize a cell, it's value can be anything.
Sometimes it happens to be "NAN", which unfortunately, compares FALSE to itself.
The solution is simply to init all values.

jpfeuffer commented on doc/doxygen/Doxyfile.in:
```diff
@@ -199,7 +199,7 @@ STRIP_FROM_INC_PATH    = @CF_OPENMS_DOCUMENTATION_STRIP_INCLUDES@
 # The default value is: NO.
 
-SHORT_NAMES            = NO
+SHORT_NAMES            = YES
```
This will break links into class documentation AFAIR

pjones replied:
Ops, I put this commit on the wrong branch.  Removed.

pjones commented:
> well, that's "easy" to fix. It seems if you do not initialize a cell, it's value can be anything. Sometimes it happens to be "NAN", which unfortunately, compares FALSE to itself. The solution is simply to init all values.

@timosachsenberg This is the test you told me to remove.  What do you think about what @cbielow found?

timosachsenberg commented:
Sure, if it works. Then it’s better for to keep it

pjones committed:
[TEST] Make the BilinearInterpolation test deterministic

The BilinearInterpolation test did not properly initialize all of the
matrix elements before comparing for equality.  Since NaN does not
compare with itself, the test would randomly fail when one or more
elements were initialized to NaN.

I also removed unused headers as reported by clangd.

pjones commented:
@timosachsenberg @jpfeuffer I fixed the test instead of removing it.
